app/experimenter/experiments/migrations/0047_auto_20190422_1819.py

```diff
@@ -0,0 +1,36 @@
+# Generated by Django 2.1.7 on 2019-04-22 18:19
+from experimenter.experiments.constants import ExperimentConstants
+from django.db import migrations
+
+
+def populate_new_risk_fields(apps, schema_editor):
+    Experiment = apps.get_model("experiments", "Experiment")
+    for experiment in Experiment.objects.all():
+        if experiment.status not in (
+            ExperimentConstants.STATUS_DRAFT,
+            ExperimentConstants.STATUS_REVIEW,
+        ):
+            for risk in all_risks():
+                setattr(
+                    experiment, risk, any([getattr(experiment, risk), False])
+                )
+                experiment.save()
```

Comment:
Ah this will do a roundtrip to the db for every field, which adds unnecessary i/o.  If you move this out of the for loop it should speed things up.

Reply:
let me fix that....

Reply:
Actually I just realized you could just do this replacing the loop body with

    Experiment.objects.filter(**{risk: None}).update(**{risk: False})

And that way it hits multiple experiments in a single query and doesn't bother hitting ones that don't need to be updated.

I don't think there's a clean way to do it for all the risk fields in a single query...

Reply:
just want to make sure i understand correctly, this would mean that I would also be getting rid of the outer loop that loops over every experiment as well?

Reply:
Yep exactly, just loop over the risk fields and execute that query once per risk and it'll hit all the experiments.

Reply:
would this mean that all experiments existing at the time of the migration will have risk values converted to False from None, regardless of state/status?

Reply:
Oh yeah you'd want to preserve the status filters, so you could either fold it into the filter part of each query, or use the fact that django querysets are lazy and go

    filtered_experiments = Experiment.objects.filter(status__in=...)
    for risk in risks:
        filtered_experiments.filter(...).update(...)

Reply:
i see i see

Reply:
extra reading for fun:

https://docs.djangoproject.com/en/2.2/topics/db/queries/#querysets-are-lazy